### PR TITLE
Hw 2 2

### PR DESCRIPTION
**Pull request by yurkohudyma**
Homework2 recommitted

### CODE REVIEW

**Review by romanbobak — CHANGES_REQUESTED** [review]
Hi Yura! Well done! Thank you for efforts and dedicated time! 
I left several minor comments mostly regarding code style. Please take a look.
Please also delete file 'application.properties' that is empty and is not used.

Also I noticed issue related to the project structure.
Lets get together via Teams and resolve it.

Thank you again for resolving remarks!

File: LabHomework2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		</dependency>
+
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-configuration-processor</artifactId>
+			<optional>true</optional>
+		</dependency>
+		
+		<dependency>
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/pom.xml:38]: I'd deleted this dep before. Seems like some version collision has taken place.

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanB.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	public void setValue(int value) {
+		this.value = value;
+	}
+	
+	/**
+	 * init method defined by XmlApplicationContext
+	 */
+    public void init() {
+        App.out("<---------Custom initMethod from BeanB---------->");
+}
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanB.java:37]: Tried both Ctrl-Alt-L and namely reformat Ctrl-Alt-Shift-L: no lines changed. What exactly is out of the style?

[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanB.java:37]: Will comply

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanC.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		return value;
+	}
+
+	public void setValue(int value) {
+		this.value = value;
+	}
+
+    public void init() {
+        System.out.println("<---------Custom initMethod from BeanC---------->");
+}
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanC.java:32]: Will do

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanD.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		return value;
+	}
+
+	public void setValue(int value) {
+		this.value = value;
+	}
+	
+    public void init() {
+        System.out.println("<---------Custom initMethod from BeanD---------->");
+}
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanD.java:33]: Affirmative

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanD.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	
+	public void destroy() {
+		System.out.println("<--------Custom destroyMethod from BeanD--------->");
+	}
+
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanD.java:38]: Done

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanE.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	
+	private String name;
+	private int value;
+	
+	
+	public String getName() {
+		return name;
+	}
+
+	public void setName() {
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanE.java:20]: Done

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanE.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+	}
+
+	public int getValue() {
+		return value;
+	}
+
+	public void setValue() {
+		}
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanE.java:28]: Done

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/config/AppConfig.java [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Lazy;
+
+@Configuration
+@Import (OtherConfig.class)
+public class AppConfig {
+	
+@Bean ("beanA")
+@DependsOn(value= {"beanD", "beanB", "beanC"})
+public BeanA getBeanA() {
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/config/AppConfig.java:23]: Done

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/extensions/BeanPostProcessorImpl.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+	public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+		App.out("\n"+beanName + ": postProcessBeforeInitialization");
+
+		if (bean instanceof Validable) {
+			Field fieldName = null;
+			Field fieldValue = null;
+			String name = null;
+			int value = 0;
+			try {
+				fieldName = bean.getClass().getDeclaredField("name");
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/extensions/BeanPostProcessorImpl.java:25]: Thanxalot!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+public class App {
+
+	
+	public static ClassPathXmlApplicationContext xmlContext = new ClassPathXmlApplicationContext("applicationContext.xml");
+	public static AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
+	
+	public static void main(String[] args) {
+
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java:15]: done

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+		
+		out("");
+		out("\n--------TASK#9-------");
+		for (String contextgetBeanDefinitionNames : context.getBeanDefinitionNames()) {
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java:19]: Done

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+		context.close();
+		xmlContext.close();
+
+	}
+
+	public static void out(String string) {
+		System.out.println(string);
+	}
+	
+	public static void outXmlContext(String beanName) {
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java:47]: Done

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanB.java [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+	 */
+    
+    public void initByBeanFactoryPostProcessor() {
+    	App.out("<---------Custom initMethod for BeanB from initByBeanFactoryPostProcessor---------->");
+    }
+	
+	public void destroy() {
+		App.out("<--------Custom destroyMethod from BeanB--------->");
+	}
+
```
[the remark this answers is not in the document]
Comment (yurkohudyma) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanB.java:50]: Done

File: LabHomework2/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/pom.xml:38]: Thank you!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanB.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanB.java:37]: Thanks!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanB.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanB.java:50]: Thanks!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanC.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanC.java:32]: Thanks!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanD.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanD.java:33]: Thanks!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanD.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanD.java:38]: Thanks!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanE.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanE.java:20]: Thanks!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanE.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/beans/BeanE.java:28]: Thanks!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/config/AppConfig.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/homework2/config/AppConfig.java:23]: Thanks!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java:15]: Thanks!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java:19]: Thanks!

File: LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (romanbobak) [reply LabHomework2/src/main/java/com/epam/spring/src/main/java/com/epam/spring/App.java:47]: Thanks!

**Pull request merged**